src/index.cpp

```diff
@@ -123,6 +118,24 @@ Index<T, TagT, LabelT>::Index(Metric m, const size_t dim, const size_t max_point
         // Note: moved this to factory, keeping this for backward compatibility.
         _data_store =
             std::make_unique<diskann::InMemDataStore<T>>((location_t)total_internal_points, _dim, this->_distance);
+
+        _pq_distance_fn = std::make_shared<PQL2Distance<T>>((uint32_t)_num_pq_chunks, use_opq);
+        // REFACTOR TODO: Unlike Distance and DataStore, where distance object is ready when the data store
+        // is constructed. Here the distance object will not be fully ready until populate_data() is called
+        _pq_data_store = std::make_shared<PQDataStore<T>>(_dim, (location_t)total_internal_points, _num_pq_chunks,
+                                                          this->_distance, _pq_distance_fn);
+        // REFACTOR
+        // alloc_aligned(
+        //     ((void **)&_pq_data), total_internal_points * _num_pq_chunks * sizeof(char), 8 * sizeof(char));
+        // std::memset(_pq_data, 0, total_internal_points * _num_pq_chunks * sizeof(char));
+    }
+
+    if (_pq_dist)
```

Comment:
Not for this method because it is present for backward compat and can be called independent of index factory. When we no longer need these "helpers" we can get rid of them and have all clients use the IndexFactory.